### PR TITLE
Pass Playlist instance as @Body to create playlist call

### PR DESCRIPTION
**Pull request by remcomokveld**
using the Query parameters caused a '400 Bad Request' since the url expects a Json body

### CODE REVIEW

File: src/main/java/kaaes/spotify/webapi/android/models/PlaylistBase.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
  * {@link kaaes.spotify.webapi.android.models.PlaylistSimple}
  */
 public abstract class PlaylistBase {
-    public boolean collaborative;
+    public Boolean collaborative;
```
Comment (kaaes) [inline src/main/java/kaaes/spotify/webapi/android/models/PlaylistBase.java]: What is the reason for this change? Is collaborative field ever null?

**Comment by thelinmichael** [comment]
Although the parameters to the Create Playlist endpoint accepts a number of properties in the Playlist object, I feel that it's incorrect for the method to accept the Playlist object as a parameter. 

1 - There are no guarantees that the parameters to the Create Playlist endpoint will continue to take only Playlist properties - or that these parameters will be given in the same form as the Playlist object. For example, the Create Playlist endpoint may take an 'image' : <string> to set the Playlist's image, which would be incompatible with the Playlist object's images array.

2 - I think it's confusing to supply a Playlist object since only a minority of its properties can actually be set on it, and it would never make sense to set its URI, ID, external URI, and so on.

I do agree that the collaborative variable should be a boxed Boolean though since it can be null.

/$0.02

**Comment by remcomokveld** [comment]
I understands your problem with the Playlist class having too many fields. I think it will be clear soon enough though that you can not set fields other then name and public since the API will give a HTTP status code 400. 

Also, what other method would you propose to send the name and public fields as a json post body? Or am i doing something else wrong that causes a 400 response whenever i try to use the method as it it currently implemented?

**Comment by thelinmichael** [comment]
@remcomokveld, I believe the current issue with using this method in this wrapper is that it's sending query parameters and not a JSON object in the request body. There's usually an error description from the Web API when it returns a 4XX though, so I'm curious about what it says.

**Comment by remcomokveld** [comment]
Yeah that is indeed the problem below is the full log. We will need to do something like the commit i proposed. Maybe with another class, but the API just really needs the json body

02-10 19:33:44.510     977-2083/nl.lowlands.debug D/Retrofit﹕ ---> HTTP POST https://api.spotify.com/v1/users/remmaa/playlists?name=Test
02-10 19:33:44.510     977-2083/nl.lowlands.debug D/Retrofit﹕ Authorization: Bearer ....
02-10 19:33:44.510     977-2083/nl.lowlands.debug D/Retrofit﹕ ---> END HTTP (no body)
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ <--- HTTP 400 https://api.spotify.com/v1/users/remmaa/playlists?name=Test (41ms)
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ Server: nginx
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ Date: Tue, 10 Feb 2015 18:33:45 GMT
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ Content-Type: application/json; charset=utf-8
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ Transfer-Encoding: chunked
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ Connection: keep-alive
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ Keep-Alive: timeout=600
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ Cache-Control: private, max-age=0
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ Access-Control-Allow-Headers: Accept, Authorization, Origin, Content-Type
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ Access-Control-Allow-Origin: *
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ Access-Control-Max-Age: 604800
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ Access-Control-Allow-Methods: GET, POST, OPTIONS, PUT, DELETE
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ Access-Control-Allow-Credentials: true
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ OkHttp-Selected-Protocol: http/1.1
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ OkHttp-Sent-Millis: 1423593224522
02-10 19:33:44.552     977-2083/nl.lowlands.debug D/Retrofit﹕ OkHttp-Received-Millis: 1423593224551
02-10 19:33:44.553     977-2083/nl.lowlands.debug D/Retrofit﹕ {
    "error": {
    "status": 400,
    "message": "Error parsing JSON."
    }
    }
02-10 19:33:44.553     977-2083/nl.lowlands.debug D/Retrofit﹕ <--- END HTTP (76-byte body)
02-10 19:33:44.660     977-2083/nl.lowlands.debug E/AndroidRuntime﹕ FATAL EXCEPTION: IntentService[RecommendationService]
    Process: nl.lowlands.debug, PID: 977
    retrofit.RetrofitError: 400 Bad Request
            at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:388)
            at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:240)
            at java.lang.reflect.Proxy.invoke(Proxy.java:397)
            at $Proxy0.createPlaylist(Unknown Source)
            at nl.lowlands.service.RecommendationService.handleSavePlaylistToSpotify(RecommendationService.java:151)
            at nl.lowlands.service.RecommendationService.onHandleIntent(RecommendationService.java:80)
            at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.os.HandlerThread.run(HandlerThread.java:61)

**Comment by thelinmichael** [comment]
@remcomokveld, this was fixed by #25 that was just merged. Please have a look!
